manticore/platforms/linux.py

```diff
@@ -1292,12 +1292,12 @@ def sys_write(self, fd, buf, count):
             data = self._transform_write_data(data)
             write_fd.write(data)
 
-            for line in ''.join([str(x) for x in data]).split('\n'):
+            for line in b''.join([x for x in data]).split(b'\n'):
```

Comment:
data can be a list of bytes, but it can also contain Expression objects (ArraySelect) instances, which i believe is why the str() call was there. so we i think we still need to do something like `str(x).encode()`

one way to easily set this up:
```
import manticore
m = manticore.Manticore('/bin/ls')
s = m.initial_state
cpu = s.cpu
arr = s.new_symbolic_buffer(4)
cpu.push_bytes(arr)
cpu.read_bytes(cpu.RSP, 4)
b = cpu.read_bytes(cpu.RSP, 4)
```

Reply:
won't `data = self._transform_write_data(data)` line 1292 directly above only return concrete data?

Also, seems there is a bug in `SLinux._transform_write_data` where `concrete_data` should be bytes()

Reply:
> won't data = self._transform_write_data(data) line 1292 directly above only return concrete data

ah you're totally right, sorry about that.

> Also, seems there is a bug in SLinux._transform_write_data where concrete_data should be bytes()

perhaps it should be bytes, i think these functions were all generally written to take/return lists. for example when `write_fd.write(data)` happens, File.write is written to loop over the argument, rather than just passing it into file.write().

Reply:
since data is guaranteed to be concrete, instead of `b''.join([x for x in data])` i think we can do `b''.join(data)`

Reply:
> data should already guaranteed to be bytes we can just use data.split(b'\n')

i think data is a list of bytes, not bytes




> I should probably review / test all the read/write syscalls to see if anything else shows up

this would be amazing 🙏

Reply:
> i think data is a list of bytes, not bytes

Ah... local changes I already made I've switched this to using `bytes`. Any objection to that? There are very few references that use it internally and this just seems to make more sense.

Reply:
updated and tested read/readv/write/writev, everything appears to be working as expected